File: csharp/FrozenPizza/FrozenPizzaService.cs
namespace FrozenPizza;

public class FrozenPizzaService
{
    private readonly IDictionary<string, int> _pizzaStock;

    public FrozenPizzaService(IDictionary<string, int> pizzaStock)
    {
        _pizzaStock = pizzaStock;
    }

    public Order TakeOrder(string name, int quantity)
    {
        if (_pizzaStock.ContainsKey(name))
        {
            var stock = _pizzaStock[name];
            if (stock >= quantity)
            {
                _pizzaStock[name]--;
                return new Order(name, quantity);
            }
            else
            {
                throw new PizzaOutOfStockException("Pizza out of stock");
            }
        }
        else
        {
            throw new PizzaNotFoundException("Pizza not found");
        }
    }
}

File: java/frozenPizza/src/main/java/FrozenPizzaService.java
import java.util.Map;

public class FrozenPizzaService {
    private final Map<String, Integer> pizzaStock;

    public FrozenPizzaService(Map<String, Integer> pizzaStock) {
        this.pizzaStock = pizzaStock;
    }

    public Order takeOrder(String name, Integer quantity) {
        if (pizzaStock.containsKey(name)) {
            Integer stock = pizzaStock.get(name);
            if (stock >= quantity) {
                pizzaStock.put(name, stock - quantity);
                return new Order(name, quantity);
            } else {
                throw new PizzaOutOfStockException("Insufficient stock");
            }
        } else {
            throw new PizzaNotFoundException("Pizza not found");
        }
    }
}
